### PR TITLE
Sa 252 comparison menu

### LINKED ISSUE #252: Implement comparison "shopping cart"
Separately from implementing the comparison page itself, the "shopping cart" for building a comparison needs to be implemented. [Figma link](https://www.figma.com/file/yaKUmuZM9xPQ1Qq5YVf5Yu/Design---Skills?node-id=3674%3A30916) plus PNGs attached below.

- The comparison menu/popover should open every time something is added to it.
- The comparison menu/popover should close if a user clicks outside of it.
- Clicking the "Compare" button should open/close the compare menu/popover.
- The "Compare" button's title should include the count of the list if it's > 0, e.g. "Compare", "Compare 1", etc.
- Clicking either of the "Add to comparison" buttons (map popups or at the top right of the detail page) should add to the list.
- Clicking an entry's delete icon should remove it from the list.
- If there are 4 entries in the comparison, the "Add to comparison" buttons should be disabled. That's the maximum we're allowing for now.
- When on the explore page, mousing over and clicking on an entry should work the same as if it was hovered as a search result.
- When on the detail page, mousing over an entry should do nothing, but clicking on an entry should navigate to that entry's detail page.

### PR DESCRIPTION
**Pull request by torosyaneric**
Resolves #252

### CODE REVIEW

File: frontend/src/components/MapPopupContent/MapPopupContent.js [hunk 1]
```diff
@@ -14,7 +18,42 @@ const StyledTitleLink = styled(Link)({
...
+          include_neighbors: true,
+          format: 'json',
+        },
+        success: geoJson => {
+          addToComparison({ id, title, bbox: geoJson?.bbox, features: geoJson?.features })
+          openCompareMenuOpen()
+        },
+      }),
+    )
+  }, [addToComparison, datasetId, dispatch, id, openCompareMenuOpen, title])
```
Comment (justinanderson) [inline frontend/src/components/MapPopupContent/MapPopupContent.js:55]: Adding a feature shouldn't wait until the geometry endpoint has responded successfully. The request might fail or take a long time to respond.

Also, all we should really store here is `id` and `title`. I wouldn't store `bbox` or `features`. We don't know how large the geometry response will be, and some uses of the Atlas might involve feature geometry which changes often enough that we won't want to cache stale versions.

**Review by justinanderson — CHANGES_REQUESTED** [review]

File: frontend/src/components/ComparisonMenu/ComparisonMenu.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            variant={'text'}
+            color={'initial'}
+            display={'flex'}
+            alignItems={'center'}
+            p={1}
+            onClick={() => onSelect(item)}
+            onMouseEnter={() => onHighlightChange(item)}
+            onMouseLeave={() => onHighlightChange(null)}>
+            <Box sx={{ width: 64 }}>
+              <StaticMap square areaId={item?.id} geoJSON={item} />
```
[the remark this answers is not in the document]
Comment (torosyaneric) [reply frontend/src/components/ComparisonMenu/ComparisonMenu.js:49]: @justinanderson I think backend does not support fetching geoJson for multiple feature IDs yet:
` # TODO: Support requests for multiple ids`
`feature_id = feature_ids.split(" ")[0]`
 in `backend/api/views.py` `geometry_for_ids`
 
 Shall I make multiple queries for now?

[the remark this answers is not in the document]
Comment (justinanderson) [reply frontend/src/components/ComparisonMenu/ComparisonMenu.js:49]: Yes, please.

**Comment by torosyaneric** [comment]
Fixed fetching issue. @justinanderson ready for review

**Review by justinanderson — CHANGES_REQUESTED** [review]
A lot better, but there's some odd behavior on Safari which was present in the previous commits as well. If the comparison menu opens because something was added to it (or if it's open and you mouse over one of the rows) then the menu jumps to the left and stays there. I'm still looking into what's going on with it, but if you have access to a machine with Safari you might have better luck diagnosing the issue.

Before mouseover:
Screen Shot 2022-04-25 at 12 02 51 PM

After mouseover:
Screen Shot 2022-04-25 at 12 02 41 PM

**Pull request merged**